components/component.go

```diff
@@ -80,7 +79,7 @@ type ManifestsConfig struct {
 }
 
 type ComponentInterface interface {
-	ReconcileComponent(ctx context.Context, cli client.Client, resConf *rest.Config, owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, currentComponentStatus bool) error
+	ReconcileComponent(ctx context.Context, cli client.Client, owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, currentComponentStatus bool) error
```

Comment:
I find it a bit puzzling to have `currentComponentStatus bool` as a contract to all components where it's only used for one component and one case. I think we should simplify it.

Reply:
i agree with you.
it was a bit hard to understand when it was introduced, we can have a follow up PR

Reply:
We came up with the fix with @aslakknutsen - see #896